mlflow/langchain/langchain_tracer.py

```diff
@@ -457,3 +492,100 @@ def flush_tracker(self):
             self._reset()
         except Exception as e:
             _logger.debug(f"Failed to flush MLflow tracer due to error {e}.")
+
+
+@dataclass
+class Chunk:
+    chunk_id: Optional[str] = None
+    doc_uri: Optional[str] = None
+    content: Optional[str] = None
+
+
+def convert_retriever_outputs(
+    documents: List[Document],
+) -> Dict[str, List[Dict[str, str]]]:
+    """
+    Convert Retriever outputs format from
+    [
+        Document(page_content="...", metadata={...}, type="Document"),
+        ...
+    ]
+    to
+    {
+        "chunks": [
+            {
+                "chunk_id": "...",
+                "doc_uri": "...",
+                "content": "...",
+            },
+            ...
+        ]
+    }
+    """
+    chunk_id_col = (
+        get_databricks_vector_search_key(DATABRICKS_VECTOR_SEARCH_PRIMARY_KEY) or VS_INDEX_ID_COL
+    )
+    doc_uri_col = (
+        get_databricks_vector_search_key(DATABRICKS_VECTOR_SEARCH_DOC_URI) or VS_INDEX_DOC_URL_COL
+    )
+    return {
+        "chunks": [
+            asdict(
+                Chunk(
+                    chunk_id=doc.metadata.get(chunk_id_col),
+                    doc_uri=doc.metadata.get(doc_uri_col),
+                    content=doc.page_content,
+                )
+            )
+            for doc in documents
+        ]
+    }
+
+
+# TODO: multiple prompts is not supported for now
+# we should update once rag eval supports it.
+def convert_llm_inputs(
+    prompts: List[str],
+) -> Dict[str, str]:
+    """
+    Convert LLM inputs format from
+    ["prompt1"...]
+    to
+    {"prompt": "prompt1"}
+    """
+    if len(prompts) == 0:
+        prompt = ""
+    elif len(prompts) == 1:
+        prompt = prompts[0]
+    else:
+        raise NotImplementedError(f"Multiple prompts not supported yet. Got: {prompts}")
+    return {"prompt": prompt}
+
+
+# TODO: This assumes we are not in batch situation where we have multiple generations per
+# input, we should handle that case once rag eval supports it.
+def convert_llm_outputs(
+    outputs: Dict[str, Any],
+) -> Dict[str, Any]:
+    """
+    Convert LLM outputs format from
+    {
+        # generations is List[List[Generation]] because
+        # each input could have multiple candidate generations.
+        "generations": [[{"text": "...", "generation_info": {...}, "type": "Generation"}]],
+        "llm_output": {...},
+        "run": [{"run_id": "..."}],
+    }
```

Comment:
```suggestion
    {
        # generations is List[List[Generation]] because
        # each input could have multiple candidate generations.
        "generations": [
            [
                {
                    "text": "...",
                    "generation_info": {...},
                    "type": "Generation",
                },
            ]
        ],
        "llm_output": {...},
        "run": [{"run_id": "..."}],
    }
```

for readability

Reply:
not a blocker